cmd/geth/dbcmd.go

```diff
@@ -810,6 +820,74 @@ func dbDelete(ctx *cli.Context) error {
 	return nil
 }
 
+// dbDeleteTrieState deletes all trie state related key-value pairs from the database and the ancient state store.
+func dbDeleteTrieState(ctx *cli.Context) error {
+	if ctx.NArg() > 0 {
+		return fmt.Errorf("no arguments required")
+	}
+
+	stack, config := makeConfigNode(ctx)
+	defer stack.Close()
+
+	db := utils.MakeChainDatabase(ctx, stack, false, false)
+	defer db.Close()
+
+	var err error
+	if db.StateStore() != nil { // if separate trie db exists, delete all files in the db folder
+		statePath := filepath.Join(stack.ResolvePath("chaindata"), "state")
+		err := filepath.Walk(statePath, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if path != statePath {
+				fileInfo, err := os.Lstat(path)
+				if err != nil {
+					return err
+				}
+				if !fileInfo.IsDir() {
+					os.Remove(path)
+				}
+			}
+			return nil
+		})
+		return err
+	} else {
+		err = rawdb.DeleteTrieState(db)
```

Comment:
it is better to delete the kv data in state by API `if db.StateStore() != nil  { 
  1)err = rawdb.DeleteTrieState(db.db.StateStore())
   2)...remove the ancient state
} else {
 err = rawdb.DeleteTrieState(db)
}`

Reply:
Changed.

Reply:
No need to revert KV data one by one if it's separate db. Reverted.